algoliasearch/client.go

```diff
@@ -243,6 +243,92 @@ func (c *client) MultipleQueriesWithRequestOptions(queries []IndexedQuery, strat
 	return
 }
 
+func (c *client) ListClusters() (res []Cluster, err error) {
+	return c.ListClustersWithRequestOptions(nil)
+}
+
+func (c *client) ListClustersWithRequestOptions(opts *RequestOptions) (res []Cluster, err error) {
+	var tmp map[string][]Cluster
+	err = c.request(&tmp, "GET", "/1/clusters", nil, read, opts)
+	res = tmp["clusters"]
+	return
+}
+
+func (c *client) ListUserIDs(page int, hitsPerPage int) (res listUserIDsRes, err error) {
+	return c.ListUserIDsWithRequestOptions(page, hitsPerPage, nil)
+}
+
+func (c *client) ListUserIDsWithRequestOptions(page int, hitsPerPage int, opts *RequestOptions) (res listUserIDsRes, err error) {
+	params := Map{
+		"page":        page,
+		"hitsPerPage": hitsPerPage,
+	}
+	err = c.request(&res, "GET", "/1/clusters/mapping?"+encodeMap(params), nil, read, opts)
+	return
+}
+
+func (c *client) GetUserID(userID string) (res UserID, err error) {
+	return c.GetUserIDWithRequestOptions(userID, nil)
+}
+
+func (c *client) GetUserIDWithRequestOptions(userID string, opts *RequestOptions) (res UserID, err error) {
+	err = c.request(&res, "GET", "/1/clusters/mapping/"+url.QueryEscape(userID), nil, read, opts)
+	return
+}
+
+func (c *client) AssignUserID(userID string, clusterName string) (res AssignUserIDRes, err error) {
+	return c.AssignUserIDWithRequestOptions(userID, clusterName, nil)
+}
+
+func (c *client) AssignUserIDWithRequestOptions(userID string, clusterName string, opts *RequestOptions) (res AssignUserIDRes, err error) {
+	if opts == nil {
+		opts = &RequestOptions{
+			ExtraHeaders: make(map[string]string),
+		}
+	}
+	opts.ExtraHeaders["X-Algolia-User-ID"] = userID
```

Comment:
Following the previous comment, do we want to check for `userID` emptiness before sending the actual call? WDYT? If not, let's make sure it's not empty by early returning the error + adding a test.